apps/laboratory/src/utils/BitcoinUtil.ts

```diff
@@ -113,6 +116,31 @@ export const BitcoinUtil = {
 
   getBitcoinNetwork(networkId: CaipNetworkId): bitcoin.Network {
     return this.isTestnet(networkId) ? bitcoin.networks.testnet : bitcoin.networks.bitcoin
+  },
+
+  getPaymentByAddress(
+    address: string,
+    network: bitcoin.networks.Network
+  ): bitcoin.payments.Payment {
+    const output = bitcoin.address.toOutputScript(address, network)
+
+    if (bitcoinPSBTUtils.isP2MS(output)) {
+      return bitcoin.payments.p2ms({ output, network })
+    } else if (bitcoinPSBTUtils.isP2PK(output)) {
+      return bitcoin.payments.p2pk({ output, network })
+    } else if (bitcoinPSBTUtils.isP2PKH(output)) {
+      return bitcoin.payments.p2pkh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2WPKH(output)) {
+      return bitcoin.payments.p2wpkh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2WSHScript(output)) {
+      return bitcoin.payments.p2wsh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2SHScript(output)) {
+      return bitcoin.payments.p2sh({ output, network })
+    } else if (bitcoinPSBTUtils.isP2TR(output)) {
+      return bitcoin.payments.p2tr({ output, network })
```

Comment:
maybe nit: can we do a switch or even an object mapping for this? 
eg
```
switch {
   case(bitcoinPSBTUtils.isP2MS(output)):
      ...
}
```

Not sure this works in js but wall of if looks ugly

Reply:
As far as I know this is not possible syntax wise, `switch` expects a value to match the value in each `case`. As we need to check through functions I don't think it is possible to use `switch` in this case.